terraform/deployments/elasticsearch/main.tf

```diff
@@ -0,0 +1,229 @@
+      terraform_deployment = "app-elasticsearch6"
+    }
+  }
+}
+
+locals {
+  domain = "${var.stackname}-elasticsearch6-domain"
+  subnet_ids = [
+    data.tfe_outputs.vpc.nonsensitive_values.private_subnet_ids["elasticsearch_a"],
+    data.tfe_outputs.vpc.nonsensitive_values.private_subnet_ids["elasticsearch_b"],
+    data.tfe_outputs.vpc.nonsensitive_values.private_subnet_ids["elasticsearch_c"],
+  ]
+  // master_user = "${var.service}-masteruser"
+}
+
+/*
+resource "random_password" "password" {
+  length  = 32
+  special = true
+}
+*/
+
+import {
+  to = aws_cloudwatch_log_group.opensearch_search_slow_logs
+  id = "/aws/aes/domains/${local.domain}/es6-search-logs"
+}
+
+resource "aws_cloudwatch_log_group" "opensearch_search_slow_logs" {
+  name              = "/aws/aes/domains/${local.domain}/es6-search-logs"
+  retention_in_days = 3
+}
+
+import {
+  to = aws_cloudwatch_log_group.opensearch_index_slow_logs
+  id = "/aws/aes/domains/${local.domain}/es6-index-logs"
+}
+
+resource "aws_cloudwatch_log_group" "opensearch_index_slow_logs" {
+  name              = "/aws/aes/domains/${local.domain}/es6-index-logs"
+  retention_in_days = 3
+}
+
+import {
+  to = aws_cloudwatch_log_group.opensearch_error_logs
+  id = "/aws/aes/domains/${local.domain}/es6-application-logs"
+}
+
+resource "aws_cloudwatch_log_group" "opensearch_error_logs" {
+  name              = "/aws/aes/domains/${local.domain}/es6-application-logs"
+  retention_in_days = 3
+}
+
+data "aws_iam_policy_document" "opensearch_log_publishing_policy" {
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["es.amazonaws.com"]
+    }
+
+    actions = [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "logs:PutLogEventsBatch",
+    ]
+
+    resources = [
+      "arn:aws:logs:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:log-group:/aws/aes/domains/${local.domain}/*"
+    ]
+  }
+}
+
+import {
+  to = aws_cloudwatch_log_resource_policy.opensearch_log_publishing_policy
+  id = "elasticsearch6_log_resource_policy"
+}
+
+resource "aws_cloudwatch_log_resource_policy" "opensearch_log_publishing_policy" {
+  policy_name = "elasticsearch6_log_resource_policy"
+
+  policy_document = data.aws_iam_policy_document.opensearch_log_publishing_policy.json
+}
+
+import {
+  to = aws_iam_service_linked_role.es_role
+  id = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/aws-service-role/es.amazonaws.com/AWSServiceRoleForAmazonElasticsearchService"
+}
+
+resource "aws_iam_service_linked_role" "es_role" {
+  aws_service_name = "es.amazonaws.com"
+}
+
+import {
+  to = aws_elasticsearch_domain.opensearch
+  id = local.domain
+}
+
+resource "aws_elasticsearch_domain" "opensearch" {
+  depends_on = [aws_iam_service_linked_role.es_role]
+
+  domain_name           = local.domain
+  elasticsearch_version = var.engine_version
+
+  cluster_config {
+    dedicated_master_enabled = var.dedicated_master != null
```

Comment:
Could replace this with 

```
locals {
    dedicated_master = coalesce(var.dedicated_master, {
        foo = bar
    })
}
```

Reply:
This needs a boolean value though which coalescing wont get me, it would get either the object value of the block, or a default (which could be a boolean, but the non-null case wouldn't have the right type).

Reply:
Good point!